label_studio/core/settings/base.py

```diff
@@ -674,3 +674,43 @@ def collect_versions_dummy(**kwargs):
 DATA_MANAGER_FILTER_ALLOWLIST = list(
     set(get_env_list('DATA_MANAGER_FILTER_ALLOWLIST') + ['updated_by__active_organization'])
 )
+
+if ENABLE_LS_CSP := get_bool_env('ENABLE_LS_CSP', True):
+    CSP_DEFAULT_SRC = (
+        "'self'",
+        "'report-sample'",
+    )
+    CSP_STYLE_SRC = ("'self'", "'report-sample'", "'unsafe-inline'")
+    CSP_SCRIPT_SRC = (
+        "'self'",
+        "'report-sample'",
+        "'unsafe-inline'",
+        "'unsafe-eval'",
+        'blob:',
+        'browser.sentry-cdn.com',
+        'https://*.googletagmanager.com',
+    )
+    CSP_IMG_SRC = (
+        "'self'",
+        "'report-sample'",
+        'data:',
+        'https://*.google-analytics.com',
+        'https://*.googletagmanager.com',
+        'https://*.google.com',
+    )
+    CSP_CONNECT_SRC = (
+        "'self'",
+        "'report-sample'",
+        'https://*.google-analytics.com',
+        'https://*.analytics.google.com',
+        'https://analytics.google.com',
+        'https://*.googletagmanager.com',
+        'https://*.g.double' + 'click.net',  # hacky way of suppressing codespell complaint
+        'https://*.ingest.sentry.io',
+    )
+    # Note that this will be overridden to true CSP for views that use the override_report_only_csp decorator
```

Comment:
I had to re-read this a few times. Does this mean that the header will be rewritten to Report CSP if true, or the LS_CSP_REPORT_ONLY=True will be overridden to False making it use CSP? I believe it is the latter just clarifying.

Reply:
```suggestion
    # Note that LS_CSP_REPORT_ONLY will be overridden and will use actual CSP for views that use the override_report_only_csp decorator
```

Reply:
The use of the word true was making it hard to quickly grok this note. Suggesting the above as a possible amendment.

Reply:
https://github.com/HumanSignal/label-studio/pull/5137/commits/f9d06093e608af53939579e17fd474354de52723 I had actually made a change for this locally already but forgot to push it :grin: agreed!